tools/apiview/parsers/CONTRIBUTING.md

```diff
@@ -0,0 +1,103 @@
+
+- Here are the models needed
+  ```
+  object APITreeNode
+    string Name
+    string Id
+    string Kind
+    Set<string> Tags
+    Dictionary<string, string> Properties
+    List<StructuredToken> TopTokens
+    List<StructuredToken> BottomTokens
+    List<APITreeNode> Children
+
+  object StructuredToken
+    string Value
+    string Id
+    StructuredTokenKind Kind
+    Set<string> Tags
+    Dictionary<string, string> Properties 
+    Set<string> RenderClasses 
+
+  enum StructuredTokenKind
+    Content
+    LineBreak
+    NoneBreakingSpace
+    TabSpace
+    ParameterSeparator
+    Url
+  ```
+### APITreeNode
+Each module of the API (namespace, class, method) should be its own node. Members of a module (methods, in a class), (classes in a namespace) should be added as child nodes of its parent module.
+
+Sort each node at each level of the tree by your desired property, this is to ensure that difference in node order does not result in diff.
+
+Ensure each node has an Id. The combination of `Id`, `Kind` and `SubKind` should make the node unique across all nodes in the tree. This is very important. For example a class and a method can potentally have the same Id, but the kind should diffrentiate them from each other.
+
+- `Name (n)`  : The name of the tree node which will be used as label for the API Navigation. Generally use the name of the module (class, method)
+- `Id (i)` : Id of the node, which should be unique at the node level. i.e. unique among its siblings. Use whatever existing parser is assigning to DefinitionId for the main Token of the node. Each node must have an Id.
+- `Kind (k)` : What kind of node is it. Please ensure you set a Kind for each node. Using any of the following `assembly`, `class`, `delegate`, `enum`, `interface`, `method` , `namespace`, `package`, `struct`, `type` will get you the corresponding default icons for the page navigation but feel free to use something language specific.
+- `Tags (t)` : Use this for opt in or opt out boolean properties. The currently supported tags are
+  - `Deprecated` Mark a node as deprecated
+  - `Hidden`  Mark a node as Hidden
+  - `HideFromNavigation` Indicate that anode should be hidden from the page navigation.
+  - `SkipDiff` Indicate that a node should not be used in computatation of diff.
+  - `CrossLandDefId` The cross language definitionId for the node.
+- `Properties (p)` : Use this for other properties of the node. The currently supported tags
+  - `SubKind` Similar to kind, use this to make the node more specific. e.g.   `Kind = 'Type'`, and `SubKind = 'class'` or somethign specific to you language. We also use this to make the navigation icon it will overide kind.
+  - `IconName` Use this only if you are looking to add a custom icon different from language wide defaults.
```

Comment:
What icon names are supported?

Reply:
This is a customization. You only need this for specifying custom Icons as such am not listing the iconNames